packages/editor-ui/src/views/NodeView.vue

```diff
@@ -2690,6 +2694,27 @@ export default mixins(
 
 
 		async mounted () {
+			window.addEventListener('message', async (message) => {
+				try {
+					const json = JSON.parse(message.data);
+					if (json && json.command === 'openWorkflow') {
+						try {
+							await this.importWorkflowExact(json);
+						} catch (e) {
+							if (window.top) {
+								window.top.postMessage(JSON.stringify({command: 'errorImportingWorkflow'}), '*');
```

Comment:
maybe `{command: 'error', message: 'Could not import workflow'}` would be better and more flexible structure (as in we can easily change this to fit new error types rather than having to support a new event every time)